syft/create_sbom_config.go

```diff
@@ -30,12 +31,12 @@ type CreateSBOMConfig struct {
 	ToolVersion       string
 	ToolConfiguration interface{}
 
-	packageTaskFactories           task.PackageTaskFactories // syft default + user-provided catalogers with tags to select with
-	persistentPackageTaskFactories task.PackageTaskFactories // user-provided catalogers without tags to select with
+	packageTaskFactories       task.PackageTaskFactories
+	packageCatalogerReferences []pkgcataloging.CatalogerReference
 }
 
-func DefaultCreateSBOMConfig() CreateSBOMConfig {
-	return CreateSBOMConfig{
+func DefaultCreateSBOMConfig() *CreateSBOMConfig {
```

Comment:
revisit convention for `Default*`, possibly move to `NewWithDefaults()`

Reply:
I think this would be a larger change -- there are ~40 `Default*` functions using this convention. If we want to change course on this convention we should break that into another PR.